apps/traffic lights.html

```diff
@@ -0,0 +1,204 @@
+
+      #l3 {
+        background-color: #4be397;
+      }
+
+      #l4 {
+        background-color: #faec46;
+      }
+
+      #l5 {
+        background-color: #f06933;
+      }
+
+      #l6 {
+        background-color: #30e620;
+      }
+
+      #l7 {
+        background-color: #f2050c;
+      }
+
+      #l8 {
+        background-color: #1799fc;
+      }
+
+      #l9 {
+        background-color: #7e43fa;
+      }
+
+      #l0 {
+        background-color: #fac618;
+      }
+
+      .off {
+        background-color: #bbb !important;
+      }
+    </style>
+    <title>Document</title>
+  </head>
+  <body>
+    <div class="traffic-light-box">
+      <div id="RedLight" class="light-circles"></div>
+      <div id="AmberLight" class="light-circles"></div>
+      <div id="GreenLight" class="light-circles"></div>
+    </div>
+
+    <div id="box" class="light-box">
+      <div id="l1" class="lights"></div>
+      <div id="l2" class="lights"></div>
+      <div id="l3" class="lights"></div>
+      <div id="l4" class="lights"></div>
+      <div id="l5" class="lights"></div>
+      <div id="l6" class="lights"></div>
+      <div id="l7" class="lights"></div>
+      <div id="l8" class="lights"></div>
+      <div id="l9" class="lights"></div>
+      <div id="l0" class="lights"></div>
+    </div>
+
+    <script>
+      function handleKeyUp(e) {
+        switch (e.key) {
+          case "r":
+            toggleLights("RedLight");
+            break;
+          case "a":
+            toggleLights("AmberLight");
+            break;
+          case "g":
+            toggleLights("GreenLight");
+            break;
+          case "1":
+            toggleLights("l1");
+            break;
+          case "2":
+            toggleLights("l2");
+            break;
+          case "3":
+            toggleLights("l3");
+            break;
+          case "4":
+            toggleLights("l4");
+            break;
+          case "5":
+            toggleLights("l5");
+            break;
+          case "6":
+            toggleLights("l6");
+            break;
+          case "7":
+            toggleLights("l7");
+            break;
+          case "8":
+            toggleLights("l8");
+            break;
+          case "9":
+            toggleLights("l9");
+            break;
+          case "0":
+            toggleLights("l0");
+            break;
+          default:
+            console.log("default");
+            break;
+        }
+      }
+
+      function toggleLights(lightId) {
+        const redLight = document.getElementById(lightId);
+
+        if (redLight.classList.contains("off")) {
+          redLight.classList.remove("off");
+        } else {
+          redLight.classList.add("off");
+        }
+      }
+
+      var pattern = ["l1", "l2", "l3", "l4", "l5", "l6", "l7", "l8", "l9"];
+
+      function disco(i) {
+        console.log(pattern[i]);
+        if (i < pattern.length) {
+          setTimeout(function () {
+            i++;
+            toggleLights(i);
```

Comment:
the `toggleLights()` function takes in as it's first argument `lightId`. It then uses that as the id of the element that it needs to change the class on. On this line you are passing in `i`. I assume that `i` in this instance is the index of the current iteration of the loop at this point? so `i` isn't the id of the element you need to change. Have a think about how to find the id of the element you need to change. 

As a clue  you already have found it in the function.